arborist/server/router_auth.go

```diff
@@ -16,73 +16,75 @@ import (
 // tokenReader extracts the `context.user.policies` field from a token. This
 // function is used for passing to `arborist.Engine.HandleAuthRequestBytes` so
 // that arborist can get the policies out of a JWT.
-func (server *Server) tokenReader(token string) ([]string, error) {
-	missingRequiredField := func(field string) error {
-		msg := fmt.Sprintf(
-			"failed to decode token: missing required field `%s`",
-			field,
-		)
-		server.Log.Error(msg)
-		return errors.New(msg)
-	}
-	fieldTypeError := func(field string) error {
-		msg := fmt.Sprintf(
-			"failed to decode token: field `%s` has wrong type",
-			field,
-		)
-		server.Log.Error(msg)
-		return errors.New(msg)
-	}
+func (server *Server) makeTokenReader(audiences []string) func(string) ([]string, error) {
+	return func(token string) ([]string, error) {
+		missingRequiredField := func(field string) error {
+			msg := fmt.Sprintf(
+				"failed to decode token: missing required field `%s`",
+				field,
+			)
+			server.Log.Error(msg)
+			return errors.New(msg)
+		}
+		fieldTypeError := func(field string) error {
+			msg := fmt.Sprintf(
+				"failed to decode token: field `%s` has wrong type",
+				field,
+			)
+			server.Log.Error(msg)
```

Comment:
Considering switch to glog? glog has both level log and location where the message happens